### ISSUE TITLE
Slight memory leak at auth-plug.c

### ISSUE THREAD

**Issue by auselen** [jpmens/mosquitto-auth-plug#230]
I was running valgrind and saw

```
==29710== 10 bytes in 1 blocks are definitely lost in loss record 2 of 14
==29710==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29710==    by 0x6025739: strdup (strdup.c:42)
==29710==    by 0x4E3B3C9: mosquitto_auth_plugin_init (auth-plug.c:176)
==29710==    by 0x4158C9: mosquitto_security_module_init (security.c:140)
==29710==    by 0x4042EA: main (mosquitto.c:301)
```
It looks like
```
p = strdup(backends);
```
is getting overwritten later
```
for (q = strsep(&p, ","); q && *q && (nord < NBACKENDS); q = strsep(&p, ",")) {
```
since `strsep` modifies p.

**Comment by auselen** [comment]
https://github.com/jpmens/mosquitto-auth-plug/pull/232

**Comment by jpmens** [comment]
closed via #232
